ehttp/src/multipart.rs

```diff
@@ -0,0 +1,142 @@
+/// Taken from ureq_multipart 1.1.1
+use std::fs::File;
+use std::io::{self, Read, Write};
+use std::path::Path;
+use mime::Mime;
+use rand::Rng;
+
+const BOUNDARY_LEN: usize = 29;
+
+fn opt_filename(path: &Path) -> Option<&str> {
+    path.file_name().and_then(|filename| filename.to_str())
+}
+
+fn random_alphanumeric(len: usize) -> String {
+    rand::thread_rng()
+        .sample_iter(&rand::distributions::Uniform::from(0..=9))
+        .take(len)
+        .map(|num| num.to_string())
+        .collect()
+}
+
+fn mime_filename(path: &Path) -> (Mime, Option<&str>) {
+    let content_type = mime_guess::from_path(path);
+    let filename = opt_filename(path);
+    (content_type.first_or_octet_stream(), filename)
+}
+
+#[derive(Debug)]
+pub struct MultipartBuilder {
+    boundary: String,
+    inner: Vec<u8>,
+    data_written: bool,
+}
+
+impl Default for MultipartBuilder {
+    fn default() -> Self {
+        Self::new()
+    }
+}
+
+#[allow(dead_code)]
+impl MultipartBuilder {
+    pub fn new() -> Self {
+        Self {
+            boundary: random_alphanumeric(BOUNDARY_LEN),
+            inner: Vec::new(),
+            data_written: false,
+        }
+    }
+    /// add text field
+    ///
+    /// * name field name
+    /// * text field text value
+    pub fn add_text(mut self, name: &str, text: &str) -> io::Result<Self> {
```

Comment:
I think we should remove all the `io::Result` from this file - they can never happen afaict.

They can also easily be avoided - just use `self.inner.extend(text.as_bytes());` instead of `write_all`, for instance